### ISSUE TITLE
git lfs push fails with files >128MB when HTTP/2 is used but works when HTTP/1.1 is used

### ISSUE THREAD

**Issue by sldr** [git-lfs/git-lfs#4655]
**Describe the bug**
Doing a `git lfs push origin --all` fails with the following error:
`LFS: Client error: https://XXX@dev.azure.com/XXX/...47f2s9b7612f3s8a5asd671937s7a55fsb6s51 from HTTP 413`
The following command was issued and then redo push command above and it will work:
`git config http.version HTTP/1.1` 

More people reporting the problem as a documentation question/issue at:
[Azure DevOps MicrosoftDocs Issue 4179](https://github.com/MicrosoftDocs/azure-devops-docs/issues/4179)

**To Reproduce**
1. Create a local repo with a lfs file of 200MB or more.
2. Create a new github repo and don't init it, but get the URL to push into.
3. Push your local repo into it (lfs file will not transfer).
4. Clone repo and ignore error that not all files transfered.
5. Setup a new remote in local repo to point to freshly cloned repo.
6. Do `git lfs puch <new remote> --all` to get lfs file into cloned repo.
7. Now go to cloned repo and issue `git lfs push origin --all` and you should get error.
8. Issue `git config http.version HTTP/1.1`
9. Now reissue `git lfs push origin --all` and it should work.

**Expected behavior**
No need for `git config http.version HTTP/1.1` command to get it to work.

**System environment**
```
Windows 10 Pro
git version 2.33.0.windows.2
git-lfs/2.13.3 (GitHub; windows amd64; go 1.16.2; git a5e65851)
system  diff.astextplain.textconv=astextplain
system  http.sslbackend=openssl
system  http.sslcainfo=C:/Program Files/Git/mingw64/ssl/certs/ca-bundle.crt
system  core.autocrlf=true
system  core.fscache=true
system  core.symlinks=false
system  core.editor="C:\\Program Files\\Microsoft VS Code\\Code.exe" --wait
system  credential.helper=manager-core
system  pull.rebase=false
system  init.defaultbranch=master
system  filter.lfs.clean=git-lfs clean -- %f
system  filter.lfs.smudge=git-lfs smudge -- %f
system  filter.lfs.process=git-lfs filter-process
system  filter.lfs.required=true
global  user.name=Stephen L. De Rudder
global  user.email=stuffies gone for sure
global  filter.lfs.clean=git-lfs clean -- %f
global  filter.lfs.smudge=git-lfs smudge -- %f
global  filter.lfs.process=git-lfs filter-process
global  filter.lfs.required=true
local   core.repositoryformatversion=0
local   core.filemode=false
local   core.bare=false
local   core.logallrefupdates=true
local   core.symlinks=false
local   core.ignorecase=true
local   remote.origin.url=https://stuffies gone
local   remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
local   branch.master.remote=origin
local   branch.master.merge=refs/heads/master
local   lfs.https://stuffies gone
local   lfs.https://stuffies gone
local   remote.srctmp.url=C:\stuffies gone
local   remote.srctmp.fetch=+refs/heads/*:refs/remotes/srctmp/*
local   http.version=HTTP/1.1
```

**Output of `git lfs env`**
```
git-lfs/2.13.3 (GitHub; windows amd64; go 1.16.2; git a5e65851)
git version 2.33.0.windows.2

Endpoint=https://stuffies gone
Endpoint (srctmp)=file:///C:/stuffies gone
LocalWorkingDir=C:\stuffies gone
LocalGitDir=C:\stuffies gone
LocalGitStorageDir=C:\stuffies gone
LocalMediaDir=C:\stuffies gone
LocalReferenceDirs=
TempDir=C:\stuffies gone
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=C:\stuffies gone
AccessDownload=basic
AccessUpload=basic
DownloadTransfers=basic,lfs-standalone-file
UploadTransfers=basic,lfs-standalone-file
GIT_EXEC_PATH=C:/Program Files/Git/mingw64/libexec/git-core
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

**Additional context**
I found this other question that maybe related to this [4447](https://github.com/git-lfs/git-lfs/issues/4447).

**Comment by bk2204** [comment]
Hey,

This isn't a limitation in Git LFS, but a problem in Azure DevOps.  They've asked for some special case behavior in #4211, which we chose not to implement for reasons I've outlined there.  This is something that could be fixed on their end, probably without too much difficulty, but they haven't done so yet.  Note that other implementers don't have this problem.

I would recommend that if you're still having problems, you reach out to them to see if they plan to fix it and decide on the best steps for you after that.
